### PR TITLE
Upload image s3

### PR DESCRIPTION
**Pull request by ngwattcos**
### Summary 

This pull request allows users to specify a background image for their event, without needing to add CORS headers in requests to allow front-end to interface with S3 and without having to whitelist all domains, thus improving security.

- [x] implemented UploadImageS3 view to upload images securely to S3
- [x] meshes well with existing front-end and backend API to save the S3 url of the uploaded image in the database

### Test Plan 

- Click "create an event" button. Select an image to upload and fill in required fields. Click upload.

### Notes 

- Currently, backend doesn't properly detect "userid"; thus, all images are saved in user_media/None/ directory in S3. This should not be apparent to the user and does not affect the running of the app, since the url is publicly accessible and correctly written into the database.

### CODE REVIEW

File: events_backend/app/views.py [hunk 1]
```diff
@@ -683,15 +692,100 @@ def get(self, request, mobile_id, format=None):
...
+            if (make_public_success):
+                print("successfully make public")
+            else:
+                print("failure to make public")
+        else:
+            print("failure to make upload")
+
+        return JsonResponse(
+            {
+                "potato": 123,
```
Comment (SamChou19815) [inline events_backend/app/views.py]: 🤔

File: events_backend/app/views.py [hunk 2]
```diff
@@ -683,15 +692,100 @@ def get(self, request, mobile_id, format=None):
...
+
+        S3_BUCKET = settings.AWS_STORAGE_BUCKET_NAME
+        timeString = dt.now().strftime("%Y%m%d_%H%M%S")
+
+
+        s3 = boto3.client(
+            "s3",
+            aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+            aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY
+        )
```
Comment (SamChou19815) [inline events_backend/app/views.py]: can probably be a singleton

File: events_frontend/src/components/CreateEvent.js [hunk 3]
```diff
@@ -172,47 +172,25 @@ class CreateEvent extends Component {
...
-          alert("Could not get signed URL.");
-        }
-      }
-    };
-    xhr.send();
+    const fileType = file.type.replace("image/", "");
+
+
+    if (fileType.length == file.type.length) {
+      alert("Could not upload image. Please use a different file type");
```
Comment (SamChou19815) [inline events_frontend/src/components/CreateEvent.js]: return here

File: events_backend/app/views.py (hunk 2 above)
Comment (ngwattcos) [reply events_backend/app/views.py]: That is definitely a good idea. But just curious, would concurrency affect the usage of the S3 client (race conditions)?

File: events_frontend/src/components/CreateEvent.js (hunk 3 above)
Comment (ngwattcos) [reply events_frontend/src/components/CreateEvent.js]: This is strange, this does not look like the code on my end. We switched to axios/promise-based flow (we also called a different endpoint.) This is what it looks like on my end:

Screenshot (1413)

File: events_backend/app/views.py (hunk 1 above)
Comment (ngwattcos) [reply events_backend/app/views.py]: sorry, forgot to delete that :P

**Pull request closed without merging**